azext_iot/operations/hub.py

```diff
@@ -148,6 +154,20 @@ def iot_device_list(
     return result
 
 
+def _iot_device_list(hub_name, target, edge_enabled=False, top=1000):
+    top = _process_top(top)
+    query = (
+        "select * from devices where capabilities.iotEdge = true"
+        if edge_enabled
+        else "select * from devices"
+    )
+    result = _iot_query(target, query, top)
```

Comment:
make sure you include the argument name (like target=target, etc) so that if someone screws up the order of the params, it wouldnt break everything.